sigv4.go

```diff
@@ -135,18 +158,37 @@ func (rt *sigV4RoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 	for _, header := range sigv4HeaderDenylist {
 		signReq.Header.Del(header)
 	}
-
-	headers, err := rt.signer.Sign(signReq, seeker, "aps", rt.region, time.Now().UTC())
+	creds, err := rt.creds.Retrieve(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving credentials: %w", err)
+	}
+	hash := sha256.Sum256(buf.Bytes())
```

Comment:
The only thing missing in this PR is that in case the body is empty we need to set the hash to `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855` which is the sha256 of an empty string. I don't think this affects prometheus as the body is always present, but still, we need to be fully compatible.

https://github.com/aws/aws-sdk-go/blob/57564ea051fab8ccbb8cac2aff285647a1aeefd1/aws/signer/v4/v4.go#L677